src/index.ts

```diff
@@ -704,12 +988,20 @@ export class Upload extends Pumpify {
     );
     const res = await this.authClient.request(combinedReqOpts);
     this.onResponse(res);
+    this.removeListener('error', errorCallback);
+
     return res;
   }
 
   private restart() {
-    this.emit('restart');
-    this.numBytesWritten = 0;
+    if (this.numBytesWritten) {
+      const message =
+        'Attempting to restart an upload after unrecoverable bytes have been written. Stopping as this could result in data loss.';
```

Comment:
This error message might be confusing for end users. What's actionable for them here?

Reply:
Good point - I'll recommend creating a new upload object.